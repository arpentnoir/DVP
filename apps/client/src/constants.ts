import { getVcContextAndRendererEndpoints } from './config';

export const { contextUrl, rendererUrl } = getVcContextAndRendererEndpoints();

export const APP_NAME = 'Digital Verification Platform';
export const SHORT_APP_NAME = 'DVP';
export const LOGO_ALT_TEXT = 'Australian Border Force Logo';

export const ROUTES = {
  HOME: '/',
  VERIFY: '/verify',
  VIEWER: '/viewer',
  ISSUE: '/issue',
};

export const FAIL_VC_FETCH_DECRYPT_ERR_MSG =
  'Unable to fetch and/or decrypt Verifiable Credential';

export const FAIL_CREATE_VC = 'Unable to create Verifiable Credential';
export const API_ENDPOINTS = { VERIFY: '/verify', ISSUE: '/issue' };

export const GENERIC_COO_META_DATA = {
  type: ['VerifiableCredential', 'OpenAttestationCredential'],
  '@context': [
    'https://www.w3.org/2018/credentials/v1',
    `${contextUrl}/AANZFTA-CoO.json`,
    `${rendererUrl}/AANZFTA-CoO-Context-Partial.json`,
    'https://schemata.openattestation.com/com/openattestation/1.0/OpenAttestation.v3.json',
  ],
  openAttestationMetadata: {
    template: {
      type: 'EMBEDDED_RENDERER',
      name: 'AANZFTACoO',
      url: rendererUrl,
    },
    proof: {
      type: 'OpenAttestationProofMethod',
      method: 'DID',
      value: 'did:ethr:0x4Bf4190a27A37d1677c8ADE25b53F1e22885531f',
      revocation: {
        type: 'NONE',
      },
    },
    identityProof: {
      type: 'DNS-DID',
      identifier: 'dvp.ha.showthething.com',
    },
  },
  issuer: {
    id: 'did:ethr:0x4Bf4190a27A37d1677c8ADE25b53F1e22885531f#controller',
    name: 'GoSource Pty Ltd',
    type: 'OpenAttestationIssuer',
  },
};
