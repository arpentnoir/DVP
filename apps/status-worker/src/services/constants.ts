export const OAStatusRouteName = 'oa-ocsp';
export const SVIPStatusRouteName = 'revocation-list-2020';
