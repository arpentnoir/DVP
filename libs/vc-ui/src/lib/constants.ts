export const DEFAULT_RENDERER = 'https://generic-templates.tradetrust.io';

export const DEFAULT_DEBOUNCE_TIME = 500;
export const DEFAULT_SEARCH_QUERY_LENGTH = 3;

export const DEFAULT_TABLE_ERROR_MESSAGE =
  'An error occurred while loading the data';
