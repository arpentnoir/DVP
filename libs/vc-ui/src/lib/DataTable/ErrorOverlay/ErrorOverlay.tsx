/**
* @module ErrorOverlay
* @description This module exports the ErrorOverlay component which renders
  an error message overlay.
*/
import { Stack } from '@mui/material';
import { useEffect, useRef } from 'react';
import { DEFAULT_TABLE_ERROR_MESSAGE } from '../../constants';
import { Text } from '../../Text';

/**
* @typedef IErrorOverlay
* @type {object}
* @property {string} [errorMessage] - The error message to be displayed. Defaults to
  the value of DEFAULT_TABLE_ERROR_MESSAGE.
*/
interface IErrorOverlay {
  errorMessage?: string;
}

export const errorMessageTestId = 'text:table-error-message';

/**
 * @function ErrorOverlay
 * @description A functional React component that renders an error message overlay.
 * @param {IErrorOverlay} props - The props to configure the ErrorOverlay component.
 * @returns {JSX.Element} The rendered ErrorOverlay component.
 */
export const ErrorOverlay = ({
  errorMessage = DEFAULT_TABLE_ERROR_MESSAGE,
}: IErrorOverlay) => {
  const errorRef = useRef<HTMLElement>(null);

  useEffect(() => {
    if (errorMessage && errorRef?.current) {
      errorRef.current.focus();
    }
  }, [errorMessage]);

  return (
    <Stack justifyContent="center" alignItems="center" height="100%">
      <Text ref={errorRef} data-testid={errorMessageTestId}>
        {errorMessage}
      </Text>
    </Stack>
  );
};
