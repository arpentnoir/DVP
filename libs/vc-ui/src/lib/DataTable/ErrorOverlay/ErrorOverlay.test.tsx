import { render } from '@testing-library/react';
import { DEFAULT_TABLE_ERROR_MESSAGE } from '../../constants';
import { errorMessageTestId, ErrorOverlay } from './ErrorOverlay';

const mockErrorMessage = 'Test error message';
describe('ErrorOverlay', () => {
  it('should render correctly', () => {
    const { baseElement } = render(<ErrorOverlay />);

    expect(baseElement).toMatchSnapshot();
  });

  it('should use the default error text', () => {
    const { getByTestId } = render(<ErrorOverlay />);

    expect(getByTestId(errorMessageTestId).textContent).toBe(
      DEFAULT_TABLE_ERROR_MESSAGE
    );
  });

  it('should use the error message passed in', () => {
    const { getByTestId } = render(
      <ErrorOverlay errorMessage={mockErrorMessage} />
    );

    expect(getByTestId(errorMessageTestId).textContent).toBe(mockErrorMessage);
  });

  it('should have focus on the error message', () => {
    const { getByTestId } = render(<ErrorOverlay />);

    const errorMessage = getByTestId(errorMessageTestId);

    expect(errorMessage.textContent).toBe(DEFAULT_TABLE_ERROR_MESSAGE);
    expect(errorMessage).toHaveFocus();
  });
});
