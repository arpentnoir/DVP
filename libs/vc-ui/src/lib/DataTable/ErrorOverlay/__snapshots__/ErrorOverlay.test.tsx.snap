// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorOverlay should render correctly 1`] = `
<body>
  <div>
    <div
      class="css-n3b4e1-MuiStack-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
        data-testid="text:table-error-message"
        tabindex="0"
      >
        An error occurred while loading the data
      </p>
    </div>
  </div>
</body>
`;
